lci_strategic_plugin/include/lci_strategic_plugin/lci_strategic_plugin.hpp

```diff
@@ -575,35 +575,36 @@ class LCIStrategicPlugin : public carma_guidance_plugins::StrategicPlugin
    * \param earliest_entry_time Earliest entry time calculated by the vehicle
    * \param traffic_light traffic signal the vehicle is using
-   * \return tuple of <final entry time the vehicle uses to enter the intersection, 
+   * \return tuple of <final entry time the vehicle uses to enter the intersection,
    *         is_entry_time_within_green_or_tbd: true if ET is in green or TBD (related to UC3, always set to true in UC2),
    *         in_tbd: true if ET is in TBD (false if in UC2)>
   */
-  std::tuple<rclcpp::Time, bool, bool> get_final_entry_time_and_conditions(const VehicleState& current_state, 
-                                                const rclcpp::Time& earliest_entry_time, 
+  std::tuple<rclcpp::Time, bool, bool> get_final_entry_time_and_conditions(const VehicleState& current_state,
+                                                const rclcpp::Time& earliest_entry_time,
                                                 lanelet::CarmaTrafficSignalPtr traffic_light);
 
   /**
-   * \brief Provides the scheduled entry time for the vehicle in the future. This scheduled time is the earliest possible entry time that 
-   *        is during green phase and after timestamps both that is kinematically possible and required times for vehicle in front to pass through first 
+   * \brief Provides the scheduled entry time for the vehicle in the future. This scheduled time is the earliest possible entry time that
+   *        is during green phase and after timestamps both that is kinematically possible and required times for vehicle in front to pass through first
+   *        NOTE: if no valid entry time described above, it return RED with end time in the future 03:14:07 on Tuesday, 19 January 2038.
    *
    * \param current_time Current state's ROS time
-   * \param earliest_entry_time Earliest ROS entry time into the intersection kinematically possible 
+   * \param earliest_entry_time Earliest ROS entry time into the intersection kinematically possible
    * \param signal CARMATrafficSignal that is responsible for the intersection along route
    * \param minimum_required_green_time Minimum seconds required by other vehicles in front to pass through the intersection first
    *
-   * \return Nearest ROS entry time during green phase.
+   * \return Nearest ROS entry time during green phase, and whether if tbd or not
    * NOTE: TSMO UC2: Algorithm 2. Entering time estimation algorithm for EVs in cooperation Class A when the SPaT plan is fixed-time (defined for C-ADS-equipped vehicles)
    */
 
-  rclcpp::Time get_nearest_green_entry_time(const rclcpp::Time& current_time, const rclcpp::Time& earliest_entry_time, lanelet::CarmaTrafficSignalPtr signal, double minimum_required_green_time = 0.0) const;
+  std::tuple<rclcpp::Time, bool> get_nearest_valid_entry_time(const rclcpp::Time& current_time, const rclcpp::Time& earliest_entry_time, lanelet::CarmaTrafficSignalPtr signal, double minimum_required_green_time = 0.0) const;
```

Comment:
Consider changing this to `std::optional<rclcpp::Time>` to indicate that a nearest valid entry time does not exist under the given function arguments.